.github/workflows/docker-build-publish-common.yml

```diff
@@ -0,0 +1,20 @@
+name: Docker Build & Publish
+
+# Trigger on push events to main (i.e. merges), pushing new semantic version
+# tags and all PRs # yamllint disable-line rule:line-length
+on:
+  push:
+    branches:
+      - "**"
```

Comment:
[question] this regex makes me think this workflow will be invoked on all branch pushes and not just to main. If so, here's a potentially more accurate comment:

```
# Trigger on all push events, new semantic version tags, and all PRs
```

[question] if this is invoked for every branch, is this equivalent to the following suggestion?

```suggestion
```